js/app/Visualization/UI/AnimationLibrary.js

```diff
@@ -0,0 +1,156 @@
+        content: '',
+        actionBarTemplate: '' +
+          '<div class="dijitDialogPaneActionBar" data-dojo-attach-point="actionBarNode">' +
+          '  <button data-dojo-type="dijit/form/Button" type="submit" data-dojo-attach-point="closeButton">Close</button>' +
+          '</div>'
+      });
+
+      self.container = new BorderContainer({style: "min-height: 300px; height: 100%; width: 100%; padding: 0; margin: 0;", liveSplitters: true});
+      self.dialog.addChild(self.container);
+
+      self.container.addChild(new ContentPane({region: 'top', style: 'border: none; padding: 0; padding-bottom: 10px; margin: 0; overflow: hidden;', content: '' +
+        '<input type="text" class="query" style="width: 100%;" placeholder="Search by name or tag"></input>' +
+        '<div class="search-loading">' +
+        '  <img style="width: 20px;" src="' + app.dirs.img + '/loader/spinner.min.svg">' +
+        '</div>'
+      }));
+
+      self.sourcesList = new ContentPane({region: 'center', content: '', class: 'sourcesList', style: 'border: none; padding: 0; margin: 0;'});
+      self.animationsList = new ContentPane({region: 'right', splitter: true, content: '', class: 'animationsList', style: 'border: none; padding: 0; margin: 0; width: 150px'});
+      self.container.addChild(self.sourcesList);
+      self.container.addChild(self.animationsList);
+
+      self.dialog.startup();
+
+      $(self.dialog.containerNode).find(".query").keyup(function(event) {
+        if (event.which == 13) {
+          self.performSearch($(self.dialog.containerNode).find(".query").val());
+        }
+      });
+
+      $(self.dialog.closeButton).on('click', function () {
+        self.dialog.hide();
+      });
+    },
+
+    displayAnimationLibraryDialog: function () {
+      var self = this;
+      $(self.sourcesList.containerNode).html('');
+      $(self.animationsList.containerNode).html('');
+      self.dialog.show();
+      self.performSearch();
+    },
+
+    performSearch: function (query) {
+      var self = this;
+
+      $(self.dialog.containerNode).find('.search-loading').show();
+      self.dataManager.listAvailableSources(function (err, sources) {
+        self.sources = sources;
+        var results = [];
+        if (!query) {
+          results = Object.keys(sources);
+        } else {
+          for (var name in sources) {
+            if (name.indexOf(query) != -1) {
+              results.push(name);
+            } else {
+              if (sources[name].tags) {
+                for (var i = 0; i < sources[name].tags.length; i++) {
+                  if (sources[name].tags[i].indexOf(query)) {
+                    results.push(name);
+                    break;
+                  }
+                }
+              }
+            }
+          }
+        }
+        results.sort();
+
+        $(self.sourcesList.containerNode).html('');
+        $(self.animationsList.containerNode).html('');
+
+        results.map(function (key) {
```

Comment:
`forEach` is more appropriate than `map` here, as you are doing stateful transformations of the container node.